url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/charts/Generic_ScatterPlot.java

```diff
@@ -0,0 +1,723 @@
+                //text_String = "" + x.stripTrailingZeros().toPlainString();
+                    //text_String = "" + x.round(mc).stripTrailingZeros().toString();
+                    //text_String = "" + x.stripTrailingZeros().toString();
+                    text_String = "" + Generic_BigDecimal.roundStrippingTrailingZeros(
+                            x,
+                            Generic_BigDecimal.getDecimalPlacePrecision(
+                                    x,
+                                    significantDigits),
+                            roundingMode).toString();
+                }
+                textWidth = getTextWidth(text_String);
+                xAxisMaxLabelHeight = Math.max(xAxisMaxLabelHeight, textWidth);
+                angle = Math.PI / 2;
+                writeText(
+                        text_String,
+                        angle,
+                        col - (textHeight / 3),
+                        row + scaleTickAndTextSeparation + scaleTickLength);
+//                    row + scaleTickLength + (textHeight / 3));
+            }
+        }
+        xAxisExtraHeightBottom += xAxisMaxLabelHeight;
+        //xAxisExtraHeightBottom += scaleTickAndTextSeparation + scaleTickLength + seperationDistanceOfAxisAndData;
+        xAxisExtraWidthRight += textHeight * 2;
+        xAxisExtraWidthLeft += textHeight / 2;
+        // Add the X axis label
+        setPaint(Color.BLACK);
+        text_String = getxAxisLabel();
+        textWidth = getTextWidth(text_String);
+        xAxisExtraHeightBottom += partTitleGap;
+        // Calculate if the xAxisLabel will require the imageWidth to increase.
+        // If the xAxisLable is wider than the XAxis it might be best to split 
+        // it and write it on multiple lines.  
+        int currentWidth = xAxisExtraWidthLeft + dataWidth + xAxisExtraWidthRight;
+        int endxAxisLabelPostion = dataStartCol + (dataWidth / 2) + (textWidth / 2);
+        if (endxAxisLabelPostion > currentWidth) {
+            xAxisExtraWidthRight += endxAxisLabelPostion - currentWidth;
+        }
+        drawString(
+                text_String,
+                dataStartCol + (dataWidth / 2) - (textWidth / 2),
+                row + xAxisExtraHeightBottom);
+        // Draw line on origin
+        if (isDrawOriginLinesOnPlot()) {
+            if (originRow <= dataEndRow && originRow >= dataStartRow) {
+                setPaint(Color.LIGHT_GRAY);
+                ab = new Line2D.Double(
+                        dataStartCol,
+                        originRow,
+                        dataEndCol,
+                        originRow);
+                draw(ab);
+            }
+        }
+        if (!isAddLegend()) {
+            xAxisExtraHeightBottom += (2 * textHeight);
+        }
+        result[0] = xAxisExtraWidthLeft;
+        result[1] = xAxisExtraWidthRight;
+        result[2] = xAxisExtraHeightBottom;
+        return result;
+    }
+
+    @Override
+    public int[] drawYAxis(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 60 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 365-416 into a separate method.